src/algokit/core/tasks/vanity_address.py

```diff
@@ -0,0 +1,108 @@
+import logging
+import time
+from enum import Enum
+from multiprocessing import Manager, Pool, cpu_count
+from multiprocessing.managers import DictProxy
+from multiprocessing.synchronize import Event as EventClass
+from multiprocessing.synchronize import Lock as LockBase
+from timeit import default_timer as timer
+
+import algosdk
+from algosdk import mnemonic
+
+logger = logging.getLogger(__name__)
+
+PROGRESS_REFRESH_INTERVAL_SECONDS = 5
+
+
+class MatchType(Enum):
+    START = "start"
+    ANYWHERE = "anywhere"
+    END = "end"
+
+
+def _log_progress(shared_data: DictProxy, stop_event: EventClass, start_time: float) -> None:
+    """Logs progress of address matching at regular intervals."""
+
+    last_log_time = start_time
+
+    while not stop_event.is_set():
+        if timer() - last_log_time >= PROGRESS_REFRESH_INTERVAL_SECONDS:
+            elapsed_time = timer() - start_time
+            logger.info(
+                f"Still searching for a match. Iterated over {shared_data['count']} addresses "
+                f"in {elapsed_time:.2f} seconds."
+            )
+            last_log_time = timer()
+        time.sleep(1)  # to avoid tight loop and reduce CPU usage
+
+
+def _search_for_matching_address(
+    keyword: str, match: MatchType, lock: LockBase, stop_event: EventClass, shared_data: DictProxy
+) -> None:
+    """
+    Searches for a matching address based on the specified keyword and matching criteria.
+
+    Args:
+        keyword (str): The keyword to search for in the address.
+        match (MatchType): The matching criteria for the keyword. It can be "start" to match addresses that start with
+        the keyword, "anywhere" to match addresses that contain the keyword anywhere,
+        or "end" to match addresses that end with the keyword.
+        lock (LockBase): A multiprocessing lock object to synchronize access to the shared data.
+        stop_event (EventClass): A multiprocessing event object to stop the search when a match is found.
+        shared_data (DictProxy): A multiprocessing dictionary to share data between processes.
+    """
+
+    while not stop_event.is_set():
+        private_key, address = algosdk.account.generate_account()  # type: ignore[no-untyped-call]
+        generated_mnemonic = mnemonic.from_private_key(private_key)  # type: ignore[no-untyped-call]
+
+        match_conditions = {
+            MatchType.START: address.startswith(keyword),
+            MatchType.ANYWHERE: keyword in address,
+            MatchType.END: address.endswith(keyword),
+        }
+
+        with lock:
```

Comment:
Note: I had a minor performance related comment on this line in the other PR

Reply:
@robdmoore refactored further to get rid of locking, instead there is a dedicated counter and the thread that runs the logger sums up those counters periodically